activerecord/lib/active_record/connection_handling.rb

```diff
@@ -133,6 +133,17 @@ def connects_to(database: {}, shards: {})
     #   ActiveRecord::Base.connected_to(role: :reading, shard: :shard_one_replica) do
     #     Dog.first # finds first Dog record stored on the shard one replica
     #   end
+    #
+    # Note that when a `connected_to` block returns a relation, the query will be executed by calling `return_value.load`.
+    # If building multiple relations within the block, you may be surprised to see that only the last relationship is executed
+    # against the expected role.
+    #
+    #   ApplicationRecord.connected_to(role: :reading) do
+    #     @people = Person.all # Reads from writing role.
+    #     @dogs = Dog.all # Reads from reading role.
```

Comment:
Thanks for the review (and all the code you write! 💚).

I'll update these examples to use `ActiveRecord::Base`

Reply:
I have made this update. Is there anything else I can address here?

Reply:
The example still isn't correct though. With `ActiveRecord::Base` both read from the reading role. The role has no bearing on whether the relation is loaded. So my issue is this documentation is conflating two things that aren't really related in the cause and effect described here.

Reply:
> If building multiple relations within the block, you may be surprised to see that only the last relationship is executed against the expected role.

What I mean is this above statement is false, it was only doing that because you were using granular swapping.

Reply:
Thank you for explaining; I think I understand your perspective. 

Perhaps this would be a better illustration of the behavior I wasn't expecting:

```ruby
    #   ActiveRecord::Base.connected_to(role: :my_custom_role) do
    #     @people = Person.all # Reads from the reading role; I expected it to read from my_custom_role because I built the query within this block, but the query was not actually *executed* within the block so the my_custom_role was not used.
    #     @dogs = Dog.all # Reads from my_custom_role role.
    #   end
```

or

```ruby
    #   ActiveRecord::Base.connected_to(role: :writing) do
    #     @people = Person.all # Reads from the reading role; I expected it to read from writing because I built the query within this block, but the query was not actually *executed* within the block so the my_custom_role was not used.
    #     @dogs = Dog.all # Reads from the reading role; I expected it to read from writing because I built the query within this block, but the query was not actually *executed* within the block so the my_custom_role was not used.
          Rails.logger.info("PR #38339 doesn't do anything because the return value of  a block isn't an ARel. I'm not good a role play.")
    #   end
```

Maybe this isn't actually worth documenting though?